src/sec_security_asn1kc.c

```diff
@@ -747,12 +747,6 @@ Sec_Result SecAsn1KC_Encode(Sec_Asn1KC *kc, SEC_BYTE *buf, SEC_SIZE buf_len, SEC
       *written = i2d_Sec_Asn1KC(kc, &buf);
     }
 
-    if (*written < 0)
```

Comment:
We still need to have this check.  But instead of testing the unsigned value written we should check the signed value der_len.

I think the routine should change so after line 739 we add a "else if (der_len < 0) {
         SEC_LOG_ERROR("der_encode_to_buffer failed");		
         return SEC_RESULT_FAILURE;
}

That check should be before the der_len > buf_len check.